types/mempool/simple.go

```diff
@@ -0,0 +1,93 @@
+package mempool
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	huandu "github.com/huandu/skiplist"
+)
+
+type simpleMempool struct {
+	txQueue *huandu.SkipList
+}
+
+type txKey struct {
+	nonce  uint64
+	sender string
+}
+
+func txKeyLessNonce(a, b any) int {
+	keyA := a.(txKey)
+	keyB := b.(txKey)
+
+	return huandu.Uint64.Compare(keyB.nonce, keyA.nonce)
```

Comment:
If you don't add `sender` to the comparision then 2 txs with the same nonce from different senders will overwrite each other on `Insert` which is definitely not what we want.